docs/configuration.md

```diff
@@ -138,6 +138,26 @@ telemetries: [
 | recordAllRequests | boolean | `false` | By default, only HTTP failed requests (i.e., those with network errors or status codes which are not 2xx) are recorded. When this field is `true`, the http telemetry will record all requests, including those with successful 2xx status codes. <br/><br/>This field does **does not apply** to X-Ray traces, where all requests are recorded. |
 | addXRayTraceIdHeader | boolean | `false` | By default, the `X-Amzn-Trace-Id` header will not be added to the HTTP request. This means that the client-side trace and server-side trace will **not be linked** in X-Ray or the ServiceLens graph.<br/><br/> When this field is `true`, the `X-Amzn-Trace-Id` header will be added to HTTP requests (`XMLHttpRequest` or `fetch`). **Adding the header is dangerous and you must test your application before setting this field to `true` in a production environment.** The header could cause CORS to fail or invalidate the request's signature if the request is signed with sigv4.
 
+## Resource
```

Comment:
issue: I feel like the `Performance` configuration is the correct place to put this configuration option. This would (1) give it a similar pattern to the `Errors` configuration, and (2) give users control over all PerformanceEntry .

## Performance

| Name | Type | Default | Description |
| --- | --- | --- | --- |
| eventLimit | Number | `10` | The maximum number of resources to record ... |
| ignore | Function | `(entry: PerformanceEntry) => { ... }` |  A function which accepts a [PerformanceEntry](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry) and returns a boolean that determines if the PerformanceEntry should be ignored. <br/><br/> By default, [PerformanceResourceTiming](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming) entries from browser extensions are ignored. |

Reply:
That can work. `ignore()` can  be shared between `NavigationPlugin` and `ResourcePlugin` in perfromance config, since both use the `PerformanceResourceTiming` interface. (PerformanceEntry doesn't contain many useful fields for users to ignore).

However, does that mean `eventLimit` should also be shared and implemented in `NavigationPlugin` and `WebVitalsPlugin`? Currently, performance config is only supplied to `ResourcePlugin`

Reply:
Refactored!

Reply:
I'm not sure how to ignore PerformanceEntries that come from browser extensions at the moment. I have to do some digging

Reply:
> PerformanceEntry doesn't contain many useful fields for users to ignore

PerformanceEntry has multiple subclasses, and is actively being extended. Rather than provide a configuration option for each subclass, why not provide a single one that can be overloaded?

Reply:
> Currently, only ResourcePlugin receives the performance config

Yes the other plugins will need access to the config.

> However, does that mean eventLimit should also be shared and implemented in NavigationPlugin and WebVitalsPlugin?

No, all navigation events and web vitals events should be recorded.

Reply:
> > PerformanceEntry doesn't contain many useful fields for users to ignore
> 
> PerformanceEntry has multiple subclasses, and is actively being extended. Rather than provide a configuration option for each subclass, why not provide a single one that can be overloaded?

My refactor stores ignore(performanceEntry) in performance config. And performanceEntry can be type cast as `PerformanceResourceTiming` for resource events or `PerformanceNavigationTiming` for navigation events.

Reply:
> I'm not sure how to ignore PerformanceEntries that come from browser extensions at the moment. I have to do some digging

This has been solved by just looking for a `chrome-extension*` prefix